packages/gatsby/src/schema/types/date.ts

```diff
@@ -209,19 +210,34 @@ export function isDate(value: MomentInput): boolean {
   return typeof value !== `number` && momentDate.isValid()
 }
 
-const formatDate = ({
+const formatDateCache = new GatsbyCacheLmdb({
+  name: `format-date-cache`,
+  encoding: `string`,
+}).init()
+
+const formatDate = async ({
   date,
   fromNow,
   difference,
   formatString,
   locale = `en`,
-}: IFormatDateArgs): string | number => {
+}: IFormatDateArgs): Promise<string | number> => {
   const normalizedDate = JSON.parse(JSON.stringify(date))
   if (formatString) {
-    return moment
+    const cacheKey = `${normalizedDate}-${formatString}`
```

Comment:
`locale` is also used, we should use it for cache key
```suggestion
    const cacheKey = `${normalizedDate}-${formatString}-${locale}`
```